### PR TITLE
Add options for sig figs / inclusion of trailing dec marks

### PR DESCRIPTION
**Pull request by rich-iannone**
This adds the `n_sigfig` arg to the `fmt_number()` function and the `drop_trailing_dec_mark` arg to the `fmt_number()`, `fmt_percent()`, and `fmt_currency()` functions.

The significant figures method uses a combination of the `signif()` and `formatC()` functions to ensure compliance with modern significant digit display rules.

The option to retain a decimal mark (here, with `drop_trailing_dec_mark`) is often seen in scientific literature and in statistical summary tables. If `drop_trailing_dec_mark` is changed from the default value, it displays the otherwise implicit decimal mark (even if there are no decimal figures required for display). 

Fixes: https://github.com/rstudio/gt/issues/535

### CODE REVIEW

File: R/utils_formatters.R [hunk 1]
```diff
@@ -225,23 +225,55 @@ scale_x_values <- function(x,
...
+  } else {
+    stop("The format provided isn't recognized.")
+  }
+
   x_str <-
     formatC(
       x = x,
+      format = format,
+      mode = mode,
       digits = decimals,
```
Comment (jcheng5) [inline R/utils_formatters.R:261]: `digits = digits`

File: R/utils_formatters.R [hunk 2]
```diff
@@ -225,23 +225,55 @@ scale_x_values <- function(x,
...
+    digits <- n_sigfig
+    flag <- "#"
+    drop0trailing <- FALSE
+  } else if (format == "f") {
+    mode <- "double"
+    digits <- decimals
+    flag <- ""
+    drop0trailing <- drop_trailing_zeros
+  } else if (format == "e") {
+    mode <- NULL
```
Comment (jcheng5) [inline R/utils_formatters.R:248]: Should be `mode <- "double"`

File: R/utils_formatters.R (hunk 1 above)
Comment (rich-iannone) [reply R/utils_formatters.R]: Now fixed.

File: R/utils_formatters.R (hunk 2 above)
Comment (rich-iannone) [reply R/utils_formatters.R]: Changed now.

**Pull request merged**